data_sources/base.py

```diff
@@ -0,0 +1,32 @@
+import exceptions
+from chardet.universaldetector import UniversalDetector
+from data_sources.interface import DataSourceInterface
+from typing import BinaryIO, IO
+
+
+class BaseDataSource(DataSourceInterface):
+    __NAME = 'Base data source class'
```

Comment:
It's better to not leave misleading examples in the base class. I'd suggest to use something like `'datasource_name'` as an example. Because it could be a bit difficult to pass data source name containing spaces in the CLI interface.

But if you're planning to create selector in the GUI later, then you should probably have a human readable name as well as machine readable name. So it could probably be a good idea to have two constants/properties like that.

// Type hint by the way?